aicsimageio/writers/timeseries_writer.py

```diff
@@ -0,0 +1,137 @@
+            the dimensions -- three dimensions: TYX and four dimensions: TYXC.
+        fps: int
+            Frames per second to attach as metadata.
+            Default: 24
+
+        Examples
+        --------
+        Data is the correct shape and dimension order
+
+        >>> image = dask.array.random.random((50, 100, 100))
+        ... TimeseriesWriter.save(image, "file.gif")
+
+        Data provided with current dimension order
+
+        >>> image = numpy.random.rand(100, 3, 1024, 2048)
+        ... TimeseriesWriter.save(image, "file.mkv", "TCYX")
+
+        Notes
+        -----
+        This writer can also be useful when wanting to create a timeseries image using
+        a non-time dimension. For example, creating a timeseries image where each frame
+        is a Z-plane from a source volumetric image as seen below.
+
+        >>> image = AICSImageIO("some_z_stack.ome.tiff")
+        ... TimeseriesWriter.save(
+        ...     data=image.get_image_data("ZYX", S=0, T=0, C=0),
+        ...     filepath="some_z_stack.mp4",
+        ...     # Overloading the Z dimension as the Time dimension
+        ...     # Technically not needed as it would have been assumed due to three dim
+        ...     dim_order="TYX",
+        ... )
+
+        """
+        # Check filepath
+        # Do not fully resolve to absolute because Mac is bad with mounted drives
+        filepath = Path(filepath).expanduser()
+
+        # Convert to dask array to make downstream usage of data have a consistent API
+        if isinstance(data, np.ndarray):
+            data = da.from_array(data)
+
+        # Shorthand num dimensions
+        n_dims = len(data.shape)
+
+        # Check num dimensions
+        if n_dims not in TimeseriesWriter.DIM_ORDERS:
+            raise InconsistentShapeError(
+                f"TimeseriesWriter requires that data must have either 3 or 4 "
+                f"dimensions. Provided data with {n_dims} dimensions. ({data.shape})"
+            )
+
+        # Assume dim order if not provided
+        if dim_order is None:
+            dim_order = TimeseriesWriter.DIM_ORDERS[n_dims]
+
+        # Uppercase dim order
+        dim_order = dim_order.upper()
+
+        # Check dimensions provided in the dim order string are all T, C, Y, or X
+        if any([dim not in ["T", "C", "Y", "X"] for dim in dim_order]):
+            raise InvalidDimensionOrderingError(
+                f"The dim_order parameter only accepts dimensions: "
+                f"'T', 'C', 'Y', or 'X' "
+                f"Provided dim_order string: '{dim_order}'."
+            )
+
+        # Transpose dimensions if dim_order not ready for imageio
+        if dim_order != TimeseriesWriter.DIM_ORDERS[n_dims]:
+            # Actual reshape of the data
+            data = reshape_data(
+                data,
+                given_dims=dim_order,
+                return_dims=TimeseriesWriter.DIM_ORDERS[n_dims],
+            )
```

Comment:
This....seems familiar :thinking: any way this could be refactored to unify the logic from the RGB writer?

Reply:
Yeah I thought about that too. Will pull out to a writer utils file.

Reply:
I tried to generalize this a bit but I think I will hold off until at least one more writer makes it's way into the repo to see more of a pattern. I worry about generalizing it too early and then having to redo some of it. Plus it didn't generalize too well.